web/server/go.md

```diff
@@ -0,0 +1,498 @@
+# Web Applications in Go
+
+This section is meant to contain of some of the lessons
+Truss has had with building web applications in Go.
+It borrows from the format of [Effective Go](https://golang.org/doc/effective_go.html)
+along with some lightweight "starter templates",
+to provide some basic standards and patterns we follow.
+Code samples are not meant to be "copy and paste" ready,
+but instead provide fundamental principles
+that can be incorporated into many applications, frameworks, and packages.
+
+We attempt to use plain Go code when possible,
+but package code may be incorporated or pseudocoded
+to mimic production use cases.
+
+## Application Layers
+
+Defining application layers is crucial
+to a maintainable API or web service.
+Picking responsibilities and contracts across layers
+can define how an application grows
+and how fast code iterations in a mature codebase will take.
+In Go,
+these layers will likely take form as packages.
+There is no one-size-fits-all approach to defining the contract of these packages,
+but we'll attempt to define some common patterns and inform future approaches.
+
+### Handlers
+
+The term "handler" is used to describe the layer that accepts HTTP requests
+and returns a response.
+While it may vary in implementation between different libraries,
+the term comes from the Go standard library [HTTP Package](https://golang.org/pkg/net/http/#Handler).
+
+The responsibility of a handler is to accept an HTTP request,
+delegate the work to complete the request,
+and respond based on error or success scenarios.
+Business logic, database code, etc. should not be in a handler.
+
+It should look rather declarative
+with error and success cases drawing two vertical lines via indentation.
+Here's a trimmed down example of a basic GET handler:
+
```

Comment:
I _really_ don't know if this is the right section for this... Because it might tread into more general polymorphism behavior, but I've definitely seen it incorrectly applied to `Handler/HandlerFunc` stuff...

ALL `HandlerFunc` are `Handler`s, but all `Handler`s are not `HandlerFunc`. Return signatures should rarely be `HandlerFunc` (even if that is the underlying implementation)... They should probably be `Handler`s.

(I may be doing the wild-person arm-waving thing here... Let me know if I can try to make gooder words about this...)

Reply:
I think there is some heavy room for update here in the future. I don't have a great example of a successful handler layer. On MilMove, we use go-swagger, and I'd much prefer closer to standard library language around it.

I've delved into this a little with handler wrapper patterns: https://medium.com/@matryer/the-http-handler-wrapper-technique-in-golang-updated-bc7fbcffa702, but that's about all I've gotten to.

If you want to update that, I'm sure folks would be happy :)